### PR TITLE
stats test

### PR DESCRIPTION
**Pull request by danielhochman**

### CODE REVIEW

**Comment by shailesh33** [comment]
Hmm... looks like the output is not a json format? Did this work for you?

**Comment by danielhochman** [comment]
that's the point of the test. this is broken on dev and master.

```
"redis_req_other":366,
}}}
```

**Comment by shailesh33** [comment]
I think this was fixed. Let me check it got lost

File: travis.sh [hunk 1]
```diff
@@ -84,6 +84,13 @@ sleep 10
 RESULT=$?
 echo $RESULT
 
+# check a single stats port
+curl -s localhost:22222 | python -mjson.tool > /dev/null
```
Comment (shailesh33) [inline travis.sh]: it should be curl -s http://localhost:22222/info

**Comment by ipapapa** [comment]
I tried on `dev` branch and as @shailesh33 mentioned test passes with `info`.

File: travis.sh (hunk 1 above)
Comment (danielhochman) [inline travis.sh]: fixed

**Comment by mikramulhaq** [comment]
@shailesh33 @ipapapa @danielhochman 
#348 regardless needs to go in. Below code is fundamentally wrong, what is pos[-1] here?

// eliminate the , if any
    if (pos[-1] == ',') {
        buf->len--;
        pos--;
    }

**Comment by shailesh33** [comment]
Yes you are probably right. Why do we get white space at the end?

**Comment by mikramulhaq** [comment]
It was hard to follow through the code, so I did not spend too much time investigating.

But if we are removing last comma, no harm in checking if there are any white spaces.
